src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -570,25 +571,22 @@ public void filterForCourseDashboard(List<StudentParticipation> participations,
             }
         }
 
-        // add results to participation
+        // add relevant submission with its result to participation
         if (participation != null) {
+            // find appropriate submission
```

Comment:
What is an `appropriate` submission? This needs to be part of the inline doc, I don't like the naming in general.

Reply:
I concretized the comment :-)